x/oracle/types/message_oracle.go

```diff
@@ -44,3 +75,104 @@ func (m *MsgRegisterOracle) GetSigners() []sdk.AccAddress {
 	}
 	return []sdk.AccAddress{oracleAddress}
 }
+
+var _ sdk.Msg = &MsgApproveOracleRegistration{}
+
+func NewMsgApproveOracleRegistration(approve *ApproveOracleRegistration, signature []byte) *MsgApproveOracleRegistration {
+	return &MsgApproveOracleRegistration{
+		ApproveOracleRegistration: approve,
+		Signature:                 signature,
+	}
+}
+
+func (m *MsgApproveOracleRegistration) Route() string {
+	return RouterKey
+}
+
+func (m *MsgApproveOracleRegistration) Type() string {
+	return "ApproveOracleRegistration"
+}
+
+func (m *MsgApproveOracleRegistration) ValidateBasic() error {
+	if err := m.ApproveOracleRegistration.ValidateBasic(); err != nil {
+		return err
+	}
+
+	if m.Signature == nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "signature is empty")
+	}
+
+	return nil
+}
+
+func (m *MsgApproveOracleRegistration) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(m)
+	return sdk.MustSortJSON(bz)
+}
+
+func (m *MsgApproveOracleRegistration) GetSigners() []sdk.AccAddress {
+	oracleAddress, err := sdk.AccAddressFromBech32(m.ApproveOracleRegistration.ApproverOracleAddress)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{oracleAddress}
+}
+
+func (m *ApproveOracleRegistration) ValidateBasic() error {
+	if len(m.UniqueId) == 0 {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "uniqueId is empty")
+	}
+
+	if _, err := sdk.AccAddressFromBech32(m.ApproverOracleAddress); err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "approverOracleAddress is invalid. address: %s, error: %s", m.ApproverOracleAddress, err.Error())
+	}
+
+	if len(m.TargetOracleAddress) == 0 {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "targetOracleAddress is empty")
+	}
```

Comment:
We can also check validity of `TargetOracleAddress`.

Reply:
Thank you. 
I fixed it(https://github.com/medibloc/panacea-core/pull/519/commits/574722174a5422d2d914af8b17eac91a3a8cab23)